src/overset/AssembleOversetDecoupledAlgorithm.C

```diff
@@ -32,40 +32,13 @@ void AssembleOversetDecoupledAlgorithm::execute()
   // Reset LHS/RHS for the overset constraint rows
   prepare_constraints();
 
-  const auto& fringeNodes = realm_.oversetManager_->fringeNodes_;
   const int numDof = eqSystem_->linsys_->numDof();
-  eqSystem_->linsys_->resetRows(fringeNodes, 0, numDof, 1.0, 0.0);
-
-#if 0
-  const int rank = NaluEnv::self().parallel_rank();
-  const int nDof = eqSystem_->linsys_->numDof();
-  const auto& bulk = realm_.bulk_data();
-
-  // RHS size is the same as number of DOFs
-  std::vector<double> lhs(nDof * nDof, 0.0);
-  std::vector<double> rhs(nDof, 0.0);
-  std::vector<int> scratchIds(nDof);
-  std::vector<double> scratchVals(nDof);
-  std::vector<stk::mesh::Entity> connected_nodes(1);
-
-  // Set diagonal values to 1.0
-  for (int i = 0; i < nDof; ++i)
-    lhs[i * nDof + i] = 1.0;
-
-  for (auto* oinfo: realm_.oversetManager_->oversetInfoVec_) {
-    auto node = oinfo->orphanNode_;
-    const int nodeRank = bulk.parallel_owner_rank(node);
-
-    // Only process nodes owned by this MPI rank
-    if (rank != nodeRank) continue;
-
-    // Assign the row for the fringe node
-    connected_nodes[0] = node;
-
-    // Don't use apply_coeff here as it checks for overset logic
-    eqSystem_->linsys_->sumInto(connected_nodes, scratchIds, scratchVals, rhs, lhs, __FILE__);
-  }
-#endif
+  const auto& fringeNodes = realm_.oversetManager_->ngpFringeNodes_;
+  auto* coeffApplier = eqSystem_->linsys_->get_coeff_applier();
+  Kokkos::parallel_for(
+    fringeNodes.size(), KOKKOS_LAMBDA(const size_t& i) {
+      coeffApplier->resetRows(1, &fringeNodes(i), 0, numDof, 1.0, 0.0);
+    });
```

Comment:
@alanw0 I wasn't sure what the best approach here would be:

1. Launch a `parallel_for` loop for all the available entities and then call `resetRows` for each entity, or
2. Launch `parallel_for` with 1 and then let `resetRows` loop over the entities 

I opted for the first one, as I felt that it would reset rows in parallel whereas looping from within `resetRows` will have 1 thread updating all the rows.

Reply:
Yes this is probably fine. I think the coeffApplier code assumes it is called on device, so it can't launch a parallel-for in there. So doing the parallel-for outside is probably the only option.